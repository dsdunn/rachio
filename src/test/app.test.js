import React from 'react';
import { shallow } from 'enzyme';
import App from '../components/App';
import getZones from '../fetch';

jest.mock('../fetch')

window.fetch = jest.fn(() => Promise.resolve({'all': 'good'}))

describe.only('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App/>)
  })

  it.skip('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call getZones on mount', () => {
    expect(getZones).toBeCalled();
  })
})