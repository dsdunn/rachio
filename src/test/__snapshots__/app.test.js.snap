// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Control Your Rachio Device
        </h1>,
        <p>
          Adjust time in minutes and click START to run!
        </p>,
        <section
          className="controls"
        >
          <div
            className="all-zones-control"
          >
            <div
              className="zone"
            >
              <h2
                className="zone-name"
              >
                All Zones
              </h2>
              <Timer
                editTime={[Function]}
                id="master-timer"
                startOrStopAll={[Function]}
              />
            </div>
          </div>
          <hr />
          <div
            className="zones-container"
          />
        </section>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Control Your Rachio Device",
        },
        "ref": null,
        "rendered": "Control Your Rachio Device",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Adjust time in minutes and click START to run!",
        },
        "ref": null,
        "rendered": "Adjust time in minutes and click START to run!",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="all-zones-control"
            >
              <div
                className="zone"
              >
                <h2
                  className="zone-name"
                >
                  All Zones
                </h2>
                <Timer
                  editTime={[Function]}
                  id="master-timer"
                  startOrStopAll={[Function]}
                />
              </div>
            </div>,
            <hr />,
            <div
              className="zones-container"
            />,
          ],
          "className": "controls",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="zone"
              >
                <h2
                  className="zone-name"
                >
                  All Zones
                </h2>
                <Timer
                  editTime={[Function]}
                  id="master-timer"
                  startOrStopAll={[Function]}
                />
              </div>,
              "className": "all-zones-control",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2
                    className="zone-name"
                  >
                    All Zones
                  </h2>,
                  <Timer
                    editTime={[Function]}
                    id="master-timer"
                    startOrStopAll={[Function]}
                  />,
                ],
                "className": "zone",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "All Zones",
                    "className": "zone-name",
                  },
                  "ref": null,
                  "rendered": "All Zones",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "editTime": [Function],
                    "id": "master-timer",
                    "startOrStopAll": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "zones-container",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Control Your Rachio Device
          </h1>,
          <p>
            Adjust time in minutes and click START to run!
          </p>,
          <section
            className="controls"
          >
            <div
              className="all-zones-control"
            >
              <div
                className="zone"
              >
                <h2
                  className="zone-name"
                >
                  All Zones
                </h2>
                <Timer
                  editTime={[Function]}
                  id="master-timer"
                  startOrStopAll={[Function]}
                />
              </div>
            </div>
            <hr />
            <div
              className="zones-container"
            />
          </section>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Control Your Rachio Device",
          },
          "ref": null,
          "rendered": "Control Your Rachio Device",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Adjust time in minutes and click START to run!",
          },
          "ref": null,
          "rendered": "Adjust time in minutes and click START to run!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="all-zones-control"
              >
                <div
                  className="zone"
                >
                  <h2
                    className="zone-name"
                  >
                    All Zones
                  </h2>
                  <Timer
                    editTime={[Function]}
                    id="master-timer"
                    startOrStopAll={[Function]}
                  />
                </div>
              </div>,
              <hr />,
              <div
                className="zones-container"
              />,
            ],
            "className": "controls",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="zone"
                >
                  <h2
                    className="zone-name"
                  >
                    All Zones
                  </h2>
                  <Timer
                    editTime={[Function]}
                    id="master-timer"
                    startOrStopAll={[Function]}
                  />
                </div>,
                "className": "all-zones-control",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h2
                      className="zone-name"
                    >
                      All Zones
                    </h2>,
                    <Timer
                      editTime={[Function]}
                      id="master-timer"
                      startOrStopAll={[Function]}
                    />,
                  ],
                  "className": "zone",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "All Zones",
                      "className": "zone-name",
                    },
                    "ref": null,
                    "rendered": "All Zones",
                    "type": "h2",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "editTime": [Function],
                      "id": "master-timer",
                      "startOrStopAll": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "className": "zones-container",
              },
              "ref": null,
              "rendered": Array [],
              "type": "div",
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
